### PR TITLE
chore: Update the tests to write to an test-output directory

### PR DESCRIPTION
**Pull request by joe-yeager**
## Description and Context

Updates the acceptance tests so that they run in a directory `test-output` instead of dumping files and directories in the `acceptance-tests` directory.  Adds a setup and teardown method to clean up this directory afterwards

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,4 @@ coverage
 acceptance-tests/my-project
 acceptance-tests/*.yml
 acceptance-tests/*.html
+acceptance-tests/test-output
```
Comment (joe-yeager) [inline .gitignore:20]: Ignore this new directory in the event it somehow doesn't get deleted.

File: acceptance-tests/globalSetup.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { existsSync, mkdirSync, rmSync } from 'fs';
```
Comment (joe-yeager) [inline acceptance-tests/globalSetup.ts:1]: This is a global setup module, so the setup and teardown methods will only run one time a piece

File: acceptance-tests/lib/cmd.ts [hunk 3]
```diff
@@ -49,6 +51,7 @@ export function createProcess(
       GITHUB_TOKEN: config.githubToken,
     },
     stdio: [null, null, null, 'ipc'], // This enables interprocess communication (IPC)
+    cwd: path.join(process.cwd(), config.testDir),
```
Comment (joe-yeager) [inline acceptance-tests/lib/cmd.ts:54]: This sets all of the commands we execute to run in the test directory.  So files generated should be placed within there.

File: acceptance-tests/lib/testState.ts [hunk 4]
```diff
@@ -23,14 +28,26 @@ export class TestState {
...
 
+  getTestConfigFileName() {
+    return this.getPathWithinTestDirectory(this.testConfigFileName);
+  }
+
+  getPathWithinTestDirectory(filepath: string) {
+    return path.join(this.testOutputDir, filepath);
+  }
+
+  existsInProjectFolder(filepath: string) {
```
Comment (brandenrodgers) [inline acceptance-tests/lib/testState.ts:43]: What's the word "project" referring to here? A generic project or a literal hubspot developer project?

File: acceptance-tests/lib/testState.ts [hunk 5]
```diff
@@ -6,15 +6,20 @@ import { existsSync, readFileSync, writeFileSync } from 'fs';
...
+import * as path from 'node:path';
+
+export const testOutputDir = 'test-output';
 
 export class TestState {
   public config: TestConfig;
   public cli: CLI;
   private testConfigFileName: string;
   private parsedYaml: ReturnType<yaml.load>;
+  private testOutputDir = 'test-output';
```
Comment (brandenrodgers) [inline acceptance-tests/lib/testState.ts:18]: Should this use the variable defined above? Just in case we ever change it

Comment (joe-yeager) [reply acceptance-tests/lib/testState.ts:18]: Whoops, yep.  Good catch

Comment (joe-yeager) [reply acceptance-tests/lib/testState.ts]: https://github.com/HubSpot/hubspot-cli/pull/1198/commits/e7b4c3d95d8a072afc89a652ff7853e8fd62a9c3

**Review by brandenrodgers — APPROVED** [review]
lgtm!

File: acceptance-tests/lib/testState.ts (hunk 4 above)
Comment (joe-yeager) [reply acceptance-tests/lib/testState.ts]: I went with `existsInTestOutputDirectory`

**Pull request merged**